kea-vscode/src/views/pull-request-contents/local-commit/local-file-tree-node.ts

```diff
@@ -0,0 +1,67 @@
+import * as path from "path";
+import * as vscode from "vscode";
+import { IAccountKey } from "../../../account/account";
+import { IKeaContext } from "../../../core/context";
+import { ILocalGitRepository, LocalCommit } from "../../../git/local-git-repository";
+import { CommitFile, FileComment, RepoId } from "../../../types/kea";
+import { FileTreeNode } from "../../common/file-tree-node";
+
+/**
+ * Represents a file in a local commit tree, extending FileTreeNode functionality.
+ */
+export class LocalFileTreeNode extends FileTreeNode {
```

Comment:
Create a new `BaseFileTreeNode`. Common functionality from `LocalFileTreeNode` and `FileTreeNode` should be moved there. `FileTreeNode` should be based on it.
`FileTreeNode` should become `RemoteFileTreeNode`. 

Similarly with the folder nodes.